src/modules/CannedMessageModule.cpp

```diff
@@ -225,8 +225,8 @@ void CannedMessageModule::sendText(NodeNum dest, const char *message, bool wantR
     p->decoded.payload.size = strlen(message);
     memcpy(p->decoded.payload.bytes, message, p->decoded.payload.size);
     if (moduleConfig.canned_message.send_bell) {
-        p->decoded.payload.bytes[p->decoded.payload.size - 1] = 7; // Bell character
-        p->decoded.payload.bytes[p->decoded.payload.size] = '\0';  // Bell character
+        p->decoded.payload.bytes[p->decoded.payload.size] = 7;        // Bell character
+        p->decoded.payload.bytes[p->decoded.payload.size + 1] = '\0'; // Bell character
         p->decoded.payload.size++;
```

Comment:
I don't think you can do that - cause +1 does not belong to you - need a bit of string math here :-)

Reply:
Added checks for max. payload size (depending on bell switch).